common/src/chain/config/emission_schedule.rs

```diff
@@ -0,0 +1,336 @@
+use crate::primitives::{Amount, BlockHeight};
+use std::time::Duration;
+
+// TODO Move to a separate module
+/// Represents a certain amount of MLT.
+#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Debug)]
+pub struct Mlt(Amount);
+
+impl Mlt {
+    /// Number of decimal digits used to represent MLT
+    pub const DECIMALS: u8 = 11;
+    /// Number of atoms in 1 MLT
+    pub const ATOMS_PER_MLT: u128 = 10u128.pow(Self::DECIMALS as u32);
+    /// Zero MLTs
+    pub const ZERO: Self = Self(Amount::from_atoms(0));
+    /// Maximum representable amount of MLTs
+    pub const MAX: Self = Self(Amount::MAX);
+
+    /// Construct from the number atomic units
+    pub const fn from_atoms(n: u128) -> Self {
+        Self(Amount::from_atoms(n))
+    }
+
+    /// Construct from the number of MLTs
+    pub const fn from_mlt(n: u64) -> Self {
+        // Since the argument is u64 and number of atoms in 1 MLT is <= u64::MAX,
+        // the result is guaranteed to fit into the internal representation of Amount (u128)
+        static_assertions::const_assert!(Mlt::ATOMS_PER_MLT <= u64::MAX as u128);
+        Self(Amount::from_atoms(n as u128 * Mlt::ATOMS_PER_MLT))
+    }
+
+    /// Convert the number of atoms to Amount
+    pub const fn to_amount_atoms(self) -> Amount {
+        self.0
+    }
+
+    /// Multiplication, saturated at [Mlt::MAX]
+    pub fn saturating_mul(self, rhs: u128) -> Self {
```

Comment:
I'm sorry, but this is not allowed. We're not allowed to use any saturating operations because it could lead to security vulnerabilities and it's a slippery slope. As boring and explicit as it sounds, every mathematical operation should return an Option or Result and be evaluated manually.